### ISSUE TITLE
player.controlBar is undefined

### ISSUE THREAD

**Issue by nbibler** [kmoskwiak/videojs-resolution-switcher#5]
It looks like at the time that `videoJsResolutionSwitcher` is being loaded into a player (the function is being called), the player hasn't yet defined the `controlBar` object. This results in a no method error `resolutionSwitcher` is not defined on `undefined`. It looks as though the `player.updateSrc` is being called immediately when this function is executed which is triggering the issue.

If the body of `videoJsResolutionSwitcher` is wrapped with a `player.ready(...)`, everything appears to work as intended.

Am I the only one seeing this issue?

Video.js: v5.0.0
Resolution Switcher: v0.1.1

**Comment by nbibler** [comment]
Hmm.. perhaps it's caused by using a mix of the dynamically defined vs. pre-defined sources. Registering the resolution switcher as a plugin when using pre-defined sources is the cause. I guess the example defines the videojs plugin _before_ explicitly calling updateSrc...

**Comment by philipgiuliani** [comment]
Hi! I can reproduce this problem. I have pre-defined the sources in the DOM.

**Comment by nbibler** [comment]
Yeah. It seems to be a bug mostly in the consistency of using the plugin. It functions if you set it up the correct way and use the correct invocation for the way in which the video tag is configured. If you mix the invocation methods, it errs.

**Comment by kmoskwiak** [comment]
Thanks. I will check it

**Comment by philipgiuliani** [comment]
@nbibler have you found a workaround, with leaving the sources in the DOM?

**Comment by nbibler** [comment]
Yeah. Don't use the "plugins" invocation. Instead just call the function on the player directly after it is ready.

**Comment by philipgiuliani** [comment]
Thank you! Worked fine for me :+1:

```javascript
videojs("myplayer", { controls: true }, function() {
  this.videoJsResolutionSwitcher({
    default: "480",
    dynamicLabel: true
  });
});
```

**Comment by kmoskwiak** [comment]
Should I treat this as a bug or expected behavior? Any suggestions?

**Comment by nbibler** [comment]
I _think_ this is a bug. It seems wrong to me that integrating with the plugin using the `plugins` initializer option would break everything (it causes a JS exception which halts VideoJS execution).

**Comment by filecage** [comment]
I can confirm this behaviour and agree that this is a bug for the same reasons that @nbibler mentioned.

**Comment by kmoskwiak** [comment]
fixed in 0.1.4 :) should work fine now.
